fintech-examples/fintech-ui/src/app/bank/list-transactions/list-transactions.component.ts

```diff
@@ -36,8 +38,26 @@ export class ListTransactionsComponent implements OnInit {
   private loadTransactions(): void {
     const settings = this.storageService.getSettings();
     const online = !this.storageService.isAfterRedirect() && !settings.cacheLot;
+
+    const consent: AisConsentRequest = {
+      access: {
+        allPsd2: AllPsd2Enum.ACCOUNTSWITHBALANCES
+      },
+      frequencyPerDay: settings.frequencyPerDay,
+      validUntil: settings.validUntil,
+      recurringIndicator: settings.recurringIndicator,
+      combinedServiceIndicator: settings.combinedServiceIndicator
+    };
+
     this.aisService
-      .getTransactions(this.bankId, this.accountId, settings.lot, online, settings.consentRequiresAuthentication)
+      .getTransactions(
+        this.bankId,
+        this.accountId,
+        settings.lot,
+        JSON.stringify(consent),
```

Comment:
The consent should be completely optional, so you would set it on settings screen only if needed. Something like StorageService.getConsentOrNull